monai/apps/deepgrow/transforms.py

```diff
@@ -0,0 +1,462 @@
+            dst = distance[seed]
+
+            g = np.asarray(np.unravel_index(seed, label.shape)).transpose().tolist()[0]
+            g[0] = dst[0]  # for debug
+            if dimensions == 2 or dims == 3:
+                pos_guidance.append(g)
+            else:
+                pos_guidance.append([g[0], sid, g[-2], g[-1]])
+
+        return np.asarray([pos_guidance, [default_guidance] * len(pos_guidance)])
+
+    def __call__(self, data):
+        d = dict(data)
+        sid = d.get(self.sid, None)
+        sids = d.get(self.sids, None)
+        if sids is not None:
+            if sid is None or sid not in sids:
+                sid = self.R.choice(sids, replace=False)
+        else:
+            sid = None
+        d[self.guidance] = self._apply(d[self.label], sid)
+        return d
+
+
+class AddGuidanceSignald(Transform):
+    """
+    Add Guidance signal for input image.
+
+    Based on the "guidance" points, apply gaussian to them and add them as new channel for input image.
+
+    Args:
+        image: key to the image source.
+        guidance: key to store guidance.
+        sigma: standard deviation for Gaussian kernel.
+        number_intensity_ch: channel index.
+        batched: whether input is batched or not.
+    """
+
+    def __init__(
+        self,
+        image: str = "image",
+        guidance: str = "guidance",
+        sigma: int = 2,
+        number_intensity_ch: int = 1,
+        batched: bool = False,
+    ):
+        self.image = image
+        self.guidance = guidance
+        self.sigma = sigma
+        self.number_intensity_ch = number_intensity_ch
+        self.batched = batched
+
+    def _get_signal(self, image, guidance):
+        dimensions = 3 if len(image.shape) > 3 else 2
+        guidance = guidance.tolist() if isinstance(guidance, np.ndarray) else guidance
+        if dimensions == 3:
+            signal = np.zeros((len(guidance), image.shape[-3], image.shape[-2], image.shape[-1]), dtype=np.float32)
+        else:
+            signal = np.zeros((len(guidance), image.shape[-2], image.shape[-1]), dtype=np.float32)
+
+        sshape = signal.shape
+        for i in range(len(guidance)):
+            for point in guidance[i]:
+                if np.any(np.asarray(point) < 0):
+                    continue
+
+                if dimensions == 3:
+                    p1 = max(0, min(int(point[-3]), sshape[-3] - 1))
+                    p2 = max(0, min(int(point[-2]), sshape[-2] - 1))
+                    p3 = max(0, min(int(point[-1]), sshape[-1] - 1))
+                    signal[i, p1, p2, p3] = 1.0
+                else:
+                    p1 = max(0, min(int(point[-2]), sshape[-2] - 1))
+                    p2 = max(0, min(int(point[-1]), sshape[-1] - 1))
+                    signal[i, p1, p2] = 1.0
+
+            if np.max(signal[i]) > 0:
+                signal[i] = gaussian_filter(signal[i], sigma=self.sigma)
```

Comment:
this could be replaced with a native impl., please see https://github.com/Project-MONAI/MONAI/blob/3f16c217dbe863bf38a372a509363258c67a86d1/monai/data/utils.py#L547-L549 for an example

Reply:
done